### PR TITLE
Added Pagination, Sorting, and Swagger to the BookController.

### PR DESCRIPTION
**Pull request by k0sm0naft**

### CODE REVIEW

**Review by ReamFOX — APPROVED** [review]
LGTM

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
I think more changes should be in this pr, there is no changes regarding Pagination and Sorting

File: src/main/java/ua/bookstore/online/controller/BookController.java [hunk 1]
```diff
@@ -18,40 +21,55 @@
...
+    @Operation(summary = "Create a new book", description = "Create a new book if isbn uniq")
     public BookDto createBook(@RequestBody @Valid CreateBookRequestDto bookDto) {
         return bookService.save(bookDto);
     }
 
     @PutMapping("/{id}")
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    @Operation(summary = "Update a new book", description = "Update a new book if exist")
     public BookDto updateBook(@PathVariable Long id,
             @RequestBody @Valid CreateBookRequestDto bookDto) {
```
Comment (karma-o) [inline src/main/java/ua/bookstore/online/controller/BookController.java]: Let's add request body documentation to swagger, including a description of it's fields, types etc.

File: src/main/java/ua/bookstore/online/controller/BookController.java [hunk 2]
```diff
@@ -18,40 +21,55 @@
...
+    @Operation(summary = "Return single book by id", description = "Return single book by id")
+    public BookDto getBookById(@PathVariable Long id) {
+        return bookService.getById(id);
+    }
+
+    @GetMapping
+    @ResponseStatus(HttpStatus.OK)
+    @Operation(summary = "Return list of books",
+            description = "Return list of books with pagination and sorting")
+    public List<BookDto> getAll(Pageable pageable) {
```
Comment (karma-o) [inline src/main/java/ua/bookstore/online/controller/BookController.java]: seems strange that documentation says that we are returning a page of books, but in fact, it is a list

File: src/main/java/ua/bookstore/online/controller/BookController.java [hunk 3]
```diff
@@ -18,40 +21,55 @@
...
+        return bookService.getAll(pageable);
+    }
+
+    @GetMapping("/search")
+    @ResponseStatus(HttpStatus.OK)
+    @Operation(summary = "Return filtered list of books",
+            description = "Return filtered list of books with pagination and sorting. "
+                    + "Parameters: title, author, isbn, price")
+    public List<BookDto> searchBooks(Pageable pageable, BookSearchParameters bookSearchParameters) {
```
Comment (karma-o) [inline src/main/java/ua/bookstore/online/controller/BookController.java]: same here, let's return a page of books

File: src/main/java/ua/bookstore/online/service/BookServiceImpl.java [hunk 4]
```diff
@@ -61,6 +63,9 @@ public BookDto update(Long id, CreateBookRequestDto bookRequestDto) {
 
     @Override
     public void delete(Long id) {
-        bookRepository.deleteById(id);
+        if (bookRepository.existsById(id)) {
```
Comment (karma-o) [inline src/main/java/ua/bookstore/online/service/BookServiceImpl.java]: I'd flip that `if`, to follow the fail early pattern

**Review by karma-o — CHANGES_REQUESTED** [review]
Looks good so far, left some comments, please take a look

File: src/main/java/ua/bookstore/online/service/BookServiceImpl.java (hunk 4 above)
Comment (k0sm0naft) [reply src/main/java/ua/bookstore/online/service/BookServiceImpl.java]: But I think my it is better than `!` at the beginning.
```suggestion
        if (!bookRepository.existsById(id)) {
```
Or should I write it like :
```suggestion
        if (bookRepository.existsById(id) == false) {
```

**Review by andrii-hoienko — APPROVED** [review]
Good job!

**Pull request merged**